File: kanban-board-core/src/main/java/me/ad/kanban/entity/Task.java
package me.ad.kanban.entity;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Task extends BaseEntity {

    @Column(nullable = false)
    private String name;

    @Lob
    private String description;

    @Column(nullable = false)
    private int priority;

    @ManyToOne
    @JoinColumn(name = "stage_id")
    private Stage category;

    @ManyToOne
    @JoinColumn(name = "team_id")
    private Team team;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "task")
    private Set<TaskStageChange> taskStageChangeSet = new HashSet<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPriority() {
        return priority;
    }

    public void setPriority(int priority) {
        this.priority = priority;
    }

    public Stage getCategory() {
        return category;
    }

    public void setCategory(Stage category) {
        this.category = category;
    }

    public Team getTeam() {
        return team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }

    public User getOwner() {
        return owner;
    }

    public void setOwner(User owner) {
        this.owner = owner;
    }

    public Set<TaskStageChange> getTaskStageChangeSet() {
        return taskStageChangeSet;
    }

    public void setTaskStageChangeSet(Set<TaskStageChange> taskStageChangeSet) {
        this.taskStageChangeSet = taskStageChangeSet;
    }
}

File: kanban-board-ui/src/app/board/task/task.ts
import { Stage } from '../stage/stage';
import { Team } from '../team/team';
import { User } from '../user/user';

class TaskStageChange {
    public id:string;
    public stage:Stage;
    public startTime:string;
    public endTime:string;
    public version:number;
}

export class Task {
    public id:string;
    public name:string;
    public description:string;
    public priority:number;
    public category:Stage;
    public team:Team;
    public owner:User;
    public taskStageChanges:Array<TaskStageChange> = new Array<TaskStageChange>();

    constructor(obj?:any) {
        if(obj){
            this.id = obj.id;
            this.name = obj.name;
            this.description = obj.description;
            this.priority = obj.priority;
            this.category = new Stage(obj.category);
            this.team = new Team(obj.team);
            this.owner = new User(obj.owner);
            this.taskStageChanges = obj.taskStageChanges;
        }
    }
}
